Tests/XMLCoderTests/AdvancedFeatures/SimpleChoiceTests.swift

```diff
@@ -85,4 +85,23 @@ final class SimpleChoiceTests: XCTestCase {
         let decoded = try XMLDecoder().decode([[IntOrString]].self, from: encoded)
         XCTAssertEqual(original, decoded)
     }
+    
+    func testMixedChoice() throws {
+        enum Choice: Equatable, Codable {
+            case one
+            case two(Int)
+            case three(String, value: Int)
+        }
+
+        struct Foo: Equatable, Codable {
+            var one = Choice.one
+            var two = Choice.two(10)
+            var three = Choice.three("qq", value: 20)
+        }
+        
+        let original = Foo()
+        let encoded = try XMLEncoder().encode(original, withRootKey: "container")
+        let decoded = try XMLDecoder().decode(Foo.self, from: encoded)
```

Comment:
If possible, I'd love a simple test that demonstrates the expected XML output.

Reply:
added output test